gen/bbmri_negotiator

```diff
@@ -0,0 +1,212 @@
+
+perunServicesInit::finalize;
+
+## creates users structure
+sub processGroups {
+	my $groupId = shift;
+	my $resourceId = shift;
+
+	my $roleIdValue;
+	my $groupRole;
+
+	if ($data->getGroupAttributeValue( group => $groupId, attrName => $A_COLLECTION_ID )) {
+		$roleIdValue = $data->getGroupAttributeValue( group => $groupId, attrName => $A_COLLECTION_ID );
+		$groupRole = $GROUP_ROLE_COLLECTION;
+	} elsif ($data->getGroupAttributeValue( group => $groupId, attrName => $A_BIOBANK_ID )) {
+		$roleIdValue = $data->getGroupAttributeValue( group => $groupId, attrName => $A_BIOBANK_ID );
+		$groupRole = $GROUP_ROLE_BIOBANK;
+	} elsif ($data->getGroupAttributeValue( group => $groupId, attrName => $A_NATIONAL_NODE_ID )) {
+		$roleIdValue = $data->getGroupAttributeValue( group => $groupId, attrName => $A_NATIONAL_NODE_ID );
+		$groupRole = $GROUP_ROLE_NN;
+	} elsif ($data->getGroupAttributeValue( group => $groupId, attrName => $A_NETWORK_ID )) {
+		$roleIdValue = $data->getGroupAttributeValue( group => $groupId, attrName => $A_NETWORK_ID );
+		$groupRole = $GROUP_ROLE_NETWORK;
+	} elsif ($data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_COLLECTION_ID )) {
+		$roleIdValue = $data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_COLLECTION_ID );
+		$groupRole = $GROUP_ROLE_ADM_COLLECTION;
+	} elsif ($data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_BIOBANK_ID )) {
+		$roleIdValue = $data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_BIOBANK_ID );
+		$groupRole = $GROUP_ROLE_ADM_BIOBANK;
+	} elsif ($data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_NATIONAL_NODE_ID )) {
+		$roleIdValue = $data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_NATIONAL_NODE_ID );
+		$groupRole = $GROUP_ROLE_ADM_NN;
+	} elsif ($data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_NETWORK_ID )) {
+		$roleIdValue = $data->getGroupAttributeValue( group => $groupId, attrName => $A_ADM_NETWORK_ID );
+		$groupRole = $GROUP_ROLE_ADM_NETWORK;
+	} else {
+		return;
+	}
+
+	for my $memberId ( $data->getMemberIdsForResourceAndGroup( resource => $resourceId, group => $groupId )) {
+		processUsers $groupId, $memberId, $roleIdValue, $groupRole;
+	}
+}
+
+sub initializeUser {
+	my ($uid, $memberId) = @_;
+	$userStruc->{$uid} = {};
+	my $email = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_EMAIL );
+	my $d_name = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_D_NAME );
+
+	$userStruc->{$uid}->{$u_email} = $email;
+	$userStruc->{$uid}->{$u_name} = $d_name;
+	$userStruc->{$uid}->{$u_collections} = [];
+	$userStruc->{$uid}->{$u_biobanks} = [];
+	$userStruc->{$uid}->{$u_national_nodes} = [];
+	$userStruc->{$uid}->{$u_networks} = [];
+	$userStruc->{$uid}->{$u_adm_collections} = [];
+	$userStruc->{$uid}->{$u_adm_biobanks} = [];
+	$userStruc->{$uid}->{$u_adm_national_nodes} = [];
+	$userStruc->{$uid}->{$u_adm_networks} = [];
```

Comment:
Not sure how this is supposed to work. You want to use strings as keys no? Ideally constants like $GROUP_ROLE_ADM_COLLECTION to reduce code in other methods as well